# HG changeset patch
# Date 1645797023 18000
#      Fri Feb 25 08:50:23 2022 -0500
# Branch py3.8
# Node ID da9e45edba98f9749af099faf4d92f6dd8fc27cc
# Parent  f74623daa872ed34c610436085e36328b1de8436
fix sys._base_executable for windows in a venv. It should point to sys.base_prefix / pypy.exe (issue 3323)

diff -r f74623daa872 -r da9e45edba98 pypy/interpreter/app_main.py
--- a/pypy/interpreter/app_main.py
+++ b/pypy/interpreter/app_main.py
@@ -1045,8 +1045,13 @@
     # from this point on, we are free to use all the unicode stuff we want,
     # This is important for py3k
     sys.executable = executable
-    # This may be wrong on windows using a launcher?
-    sys._base_executable = executable
+    if sys.platform == 'win32':
+        # someday PyPy will grow a PEP 397 launcher. Until then ...
+        exe = executable.replace('\\', '/').rsplit('/', 1)[-1]
+        sys._base_executable = sys.base_prefix + '\\' + exe
+    else:
+        sys._base_executable = executable
+
 
 @hidden_applevel
 def entry_point(executable, bargv, argv):

